### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by CZED0M1**
Refaktoring kódu.

Dominik Lopauer

### CODE REVIEW

File: NNPTPZ1/Program.cs [hunk 1]
```diff
@@ -23,311 +11,141 @@ namespace NNPTPZ1
...
-                        //Console.WriteLine($"{q} {ox} -({diff})");
-                        if (Math.Pow(diff.Re, 2) + Math.Pow(diff.Imaginari, 2) >= 0.5)
-                        {
-                            q--;
-                        }
-                        it++;
-                    }
+        static void Main(string[] args)
+        {
+            int[] imagineDimensions = ReadImagineDimensions(args);
```
Comment (rodi0878) [inline NNPTPZ1/Program.cs]: please find a better datatype for representing bounds of image

File: NNPTPZ1/Program.cs [hunk 2]
```diff
@@ -23,311 +11,141 @@ namespace NNPTPZ1
...
-                for (int q = 1; q < Coe.Count; q++)
+                for (int j = 0; j < imagineDimensions[1]; j++)
                 {
-                    p.Coe.Add(Coe[q].Multiply(new Cplx() { Re = q }));
+                    ComplexNumber worldCoordinateForPixel = GetWorldCoordinatesForPixel(worldCoordinates, xStep, yStep, i, j);
+                    float iterationCount = IterateNewton(polynomial, derivedPolynomial, ref worldCoordinateForPixel);
+
+                    int rootId = FindRootId(roots, ref maxid, worldCoordinateForPixel);
+
+                    var color = colors[rootId % colors.Length];
```
Comment (rodi0878) [inline NNPTPZ1/Program.cs]: color choosing is very low level in comparison to rest of method

File: NNPTPZ1/Program.cs [hunk 3]
```diff
@@ -23,311 +11,141 @@ namespace NNPTPZ1
...
         }
 
-        public class Cplx
+        private static float IterateNewton(Polynomial polynomial, Polynomial derivedPolynomial, ref ComplexNumber worldCoordinateForPixel)
         {
-            public double Re { get; set; }
-            public float Imaginari { get; set; }
-
-            public override bool Equals(object obj)
+            float iterationCount = 0;
```
Comment (rodi0878) [inline NNPTPZ1/Program.cs]: why float for count?

File: NNPTPZ1/Program.cs [hunk 4]
```diff
@@ -23,311 +11,141 @@ namespace NNPTPZ1
...
-            public Cplx Subtract(Cplx b)
-            {
-                Cplx a = this;
-                return new Cplx()
-                {
-                    Re = a.Re - b.Re,
-                    Imaginari = a.Imaginari - b.Imaginari
-                };
-            }
+        private static Polynomial createPolynomial()
```
Comment (rodi0878) [inline NNPTPZ1/Program.cs]: broken naming convention

**Review by rodi0878 — CHANGES_REQUESTED** [review]
Please update code according to mentioned problems, also i dont see definictions of mentioned other classes - ComplexNumber and Polynomial

**Comment by CZED0M1** [comment]
Dodělány požadované změny.
Dominik Lopauer

**Comment by rodi0878** [comment]
accepted

**Pull request closed without merging**